Reduce use of WTF_ALLOW_UNSAFE_BUFFER in WTF/

cdumez opened the pull request:
#### 03cb2e5592d3dd3bd3f9925ca524d66abcb24ee1
<pre>
Reduce use of WTF_ALLOW_UNSAFE_BUFFER in WTF/
<a href="https://bugs.webkit.org/show_bug.cgi?id=283775">https://bugs.webkit.org/show_bug.cgi?id=283775</a>

Reviewed by Darin Adler.

* Source/WTF/wtf/BloomFilter.h:
(WTF::BloomFilter&lt;keyBits&gt;::keysFromHash):
* Source/WTF/wtf/Brigand.h:
* Source/WTF/wtf/ParallelJobs.h:
* Source/WTF/wtf/ParallelJobsLibdispatch.h:
(WTF::ParallelEnvironment::execute):
* Source/WTF/wtf/TinyLRUCache.h:
(WTF::TinyLRUCache::get):
(WTF::TinyLRUCache::cacheBuffer):

Canonical link: <a href="https://commits.webkit.org/287158@main">https://commits.webkit.org/287158@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/143a062a3f81f4c4fa10746ebdfc72bb0192b98e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/78551 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/57596 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/31933 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83212 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/29816 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/66747 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/5877 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61529 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19446 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/81618 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51539 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/69799 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/41840 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/48885 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/25374 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28153 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/71697 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/69985 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/25747 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/84578 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/77788 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/5916 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4054 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/69754 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6077 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/67516 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69008 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13027 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/11474 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/100097 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12130 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/5863 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21865 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/5851 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9285 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/7638 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/BloomFilter.h:
```diff
@@ -105,8 +104,8 @@ inline std::pair<unsigned, unsigned> BloomFilter<keyBits>::keysFromHash(const st
-        *reinterpret_cast_ptr<const unsigned*>(hash.data()),
-        *reinterpret_cast_ptr<const unsigned*>(hash.data() + sizeof(unsigned))
+        reinterpretCastSpanStartTo<unsigned>(std::span { hash }),
+        reinterpretCastSpanStartTo<unsigned>(std::span { hash }.subspan(sizeof(unsigned)))
```
I would have put the span into a local instead of repeating the cast twice, even though the old code repeated the pointer expression twice.

cdumez committed:
Reduce use of WTF_ALLOW_UNSAFE_BUFFER in WTF/
https://bugs.webkit.org/show_bug.cgi?id=283775

Reviewed by Darin Adler.

* Source/WTF/wtf/BloomFilter.h:
(WTF::BloomFilter<keyBits>::keysFromHash):
* Source/WTF/wtf/Brigand.h:
* Source/WTF/wtf/ParallelJobs.h:
* Source/WTF/wtf/ParallelJobsLibdispatch.h:
(WTF::ParallelEnvironment::execute):
* Source/WTF/wtf/TinyLRUCache.h:
(WTF::TinyLRUCache::get):
(WTF::TinyLRUCache::cacheBuffer):

Canonical link: https://commits.webkit.org/287158@main
